domain_config_ui/domain_config_ui.services.yml

```diff
@@ -0,0 +1,16 @@
+services:
+  domain_config_ui.manager:
+    class: Drupal\domain_config_ui\DomainConfigUIManager
+    arguments: ['@config.storage', '@domain.loader', '@language_manager']
```

Comment:
I've changed it back to use the negotiator on a new branch here https://github.com/davidgrayston/domain/tree/8.x-1.x-config_ui_domain_switch_negotiator

Diff with current pull request branch: https://github.com/davidgrayston/domain/compare/8.x-1.x-config_ui_domain_switch...8.x-1.x-config_ui_domain_switch_negotiator

I had to make some changes to DomainConfigOverrider to make this work:
- Setter to allow the domain and language to be set
- Changed protected getDomainConfigName method to be public

This doesn't seem like the correct solution in its current state